src/dom-util.js

```diff
@@ -0,0 +1,238 @@
+import { Droppable } from '@shopify/draggable';
+import Ship from './ship';
+import { randomizeBoard, getRandomData } from './util';
+import Swal from 'sweetalert2'
+
+function markShot(shot) {
+  if (!shot) return;
+  let type = 'miss2';
+  if (shot.ship) {
+    type = 'boom';
+    if (shot.ship.hp === 0 && shot.board === 'opp') {
+      const ship = document.createElement('img');
+      ship.className = `ship ship-${shot.ship.length}`;
+      if (!shot.ship.horizontal) { ship.classList.add('v-ship'); }
+      ship.src = `../src/images/ship${shot.ship.length}.png`;
+      document.querySelector(`.opp[data-x="${shot.ship.head.x}"][data-y="${shot.ship.head.y}"]`).appendChild(ship);
+    }
+  }
+  const effect = document.createElement('img');
+  effect.className = 'effect';
+  effect.src = `../src/images/${type}.png`;
+  const cell = document.querySelector(`.${shot.board}[data-x="${shot.x}"][data-y="${shot.y}"]`);
+  cell.classList.remove('darkened');
+  cell.appendChild(effect);
+}
+
+function disableBoard(boardDiv) {
+  boardDiv.childNodes.forEach((e) => e.style.pointerEvents = 'none');
+}
+
+function enableBoard(boardDiv) {
+  boardDiv.childNodes.forEach((e) => e.style.pointerEvents = 'auto');
+}
+
+function createBoard(game) {
```

Comment:
[Optional] This is a very big method. Also, it's doing a lot's of the task at a time. So if it possible then break this function in small parts according to their task.